Code/Framework/AzCore/AzCore/DOM/DomPatch.h

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/DOM/DomPath.h>
+#include <AzCore/DOM/DomValue.h>
+
+namespace AZ::Dom
+{
+    //! A patch operation that represents an atomic operation for mutating or validating a Value.
+    //! PatchOperations can be created with helper methods in Patch. /see Patch
+    class PatchOperation final
+    {
+    public:
+        using PatchOutcome = AZ::Outcome<void, AZStd::string>;
+
+        //! The operation to perform.
+        enum class Type
+        {
+            Add, //!< Inserts or replaces the value at DestinationPath with Value
+            Remove, //!< Removes the entry at DestinationPath
+            Replace, //!< Replaces the value at DestinationPath with Value
+            Copy, //!< Copies the contents of SourcePath to DestinationPath
+            Move, //!< Moves the contents of SourcePath to DestinationPath
+            Test //!< Ensures the contents of DestinationPath match Value or fails, performs no mutations
+        };
+
+        PatchOperation() = default;
+        PatchOperation(const PatchOperation&) = default;
+        PatchOperation(PatchOperation&&) = default;
+
+        explicit PatchOperation(Path destinationPath, Type type, Value value);
+        explicit PatchOperation(Path destionationPath, Type type, Path sourcePath);
+        explicit PatchOperation(Path path, Type type);
```

Comment:
These functions shouldn't be marked explicit.

Also there is a typo in `destionationPath`
```suggestion
        PatchOperation(Path destinationPath, Type type, Value value);
        PatchOperation(Path destinationPath, Type type, Path sourcePath);
        PatchOperation(Path path, Type type);
```

Reply:
Fixed, thanks